docker/src/main/java/brooklyn/entity/container/docker/DockerInfrastructure.java

```diff
@@ -86,12 +86,15 @@
     @SetFromFlag("enableSdn")
     ConfigKey<Boolean> SDN_ENABLE = SdnAttributes.SDN_ENABLE;
 
+    @SuppressWarnings("serial")
     @SetFromFlag("sdnProviderSpec")
-    ConfigKey<EntitySpec> SDN_PROVIDER_SPEC = ConfigKeys.newConfigKey(EntitySpec.class, "sdn.provider.spec", "SDN provider entity specification");
+    ConfigKey<EntitySpec<?>> SDN_PROVIDER_SPEC = ConfigKeys.newConfigKey(
+            new TypeToken<EntitySpec<?>>() {}, "sdn.provider.spec", "SDN provider entity specification");
 
+    @SuppressWarnings("serial")
     @SetFromFlag("hostSpec")
-    AttributeSensorAndConfigKey<EntitySpec, EntitySpec> DOCKER_HOST_SPEC = ConfigKeys.newSensorAndConfigKey(
-            EntitySpec.class, "docker.host.spec", "Specification to use when creating child Docker Hosts",
+    AttributeSensorAndConfigKey<EntitySpec<?>, EntitySpec<?>> DOCKER_HOST_SPEC = ConfigKeys.newSensorAndConfigKey(
+            new TypeToken<EntitySpec<?>>() {}, "docker.host.spec", "Specification to use when creating child Docker Hosts",
```

Comment:
Yes, I'd prefer to leave this (and similar instances) _without_ the generic types rather than explicitly saying `Blah<?>` everywhere, and having to use `TypeToken<Blah<?>>` instead of just `Blah.class` since it adds no extra information.